### PR TITLE
[Improve][Connector-V2] Replace `CommonErrorCodeDeprecated.JSON_OPERATION_FAILED`

### PR DESCRIPTION
**Pull request by CheneyYin**
### Purpose of this pull request

Replace `CommonErrorCodeDeprecated.JSON_OPERATION_FAILED`
Related:
- #5948

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by CheneyYin** [comment]
PTAL @Hisoka-X @hailin0

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/inject/StringInjectFunction.java [hunk 1]
```diff
@@ -51,8 +50,7 @@ public void injectFields(PreparedStatement statement, int index, Object value)
                 statement.setString(index, value.toString());
             }
         } catch (JsonProcessingException e) {
-            throw new ClickhouseConnectorException(
-                    CommonErrorCodeDeprecated.JSON_OPERATION_FAILED, e.getMessage());
+            throw CommonError.jsonOperationError("Common", value.toString(), e);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/inject/StringInjectFunction.java]:
```suggestion
            throw CommonError.jsonOperationError("Clickhouse", value.toString(), e);
```

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 2]
```diff
@@ -82,65 +83,81 @@ public String serializeRow(SeaTunnelRow row) {
...
-                return new StringBuilder()
-                        .append("{ \"index\" :")
-                        .append(objectMapper.writeValueAsString(indexMetadata))
-                        .append("}")
-                        .append("\n")
-                        .append(objectMapper.writeValueAsString(document))
-                        .toString();
+            documentStr = objectMapper.writeValueAsString(document);
+        } catch (JsonProcessingException e) {
+            throw CommonError.jsonOperationError("Common", "document:" + document.toString(), e);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java]:
```suggestion
            throw CommonError.jsonOperationError("ElasticSearch", "document:" + document.toString(), e);
```

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 3]
```diff
@@ -82,65 +83,81 @@ public String serializeRow(SeaTunnelRow row) {
...
+        String indexMetadataStr;
+        try {
+            indexMetadataStr = objectMapper.writeValueAsString(indexMetadata);
         } catch (JsonProcessingException e) {
-            throw new ElasticsearchConnectorException(
-                    CommonErrorCodeDeprecated.JSON_OPERATION_FAILED,
-                    "Object json deserialization exception.",
-                    e);
+            throw CommonError.jsonOperationError(
+                    "Common", "indexMetadata:" + indexMetadata.toString(), e);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java]:
```suggestion
                    "ElasticSearch", "indexMetadata:" + indexMetadata.toString(), e);
```

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 4]
```diff
@@ -82,65 +83,81 @@ public String serializeRow(SeaTunnelRow row) {
...
-                    .append("}")
-                    .toString();
+            deleteMetadataStr = objectMapper.writeValueAsString(deleteMetadata);
         } catch (JsonProcessingException e) {
-            throw new ElasticsearchConnectorException(
-                    CommonErrorCodeDeprecated.JSON_OPERATION_FAILED,
-                    "Object json deserialization exception.",
-                    e);
+            throw CommonError.jsonOperationError(
+                    "Common", "deleteMetadata:" + deleteMetadata.toString(), e);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java]: ditto

File: seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeDataGenerator.java [hunk 5]
```diff
@@ -67,7 +68,7 @@ private SeaTunnelRow convertRow(FakeConfig.RowData rowData) {
             seaTunnelRow.setTableId(tableId);
             return seaTunnelRow;
         } catch (IOException e) {
-            throw new FakeConnectorException(CommonErrorCodeDeprecated.JSON_OPERATION_FAILED, e);
+            throw CommonError.jsonOperationError("Common", rowData.getFieldsJson(), e);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeDataGenerator.java]: ditto

File: seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/deserialize/GoogleSheetsDeserializer.java [hunk 6]
```diff
@@ -43,20 +44,20 @@ public GoogleSheetsDeserializer(
...
+            if (i < fields.length) {
+                map.put(fields[i], row.get(i));
             }
+        }
+
+        try {
             String rowStr = objectMapper.writeValueAsString(map);
             return deserializationSchema.deserialize(rowStr.getBytes());
+        } catch (JsonProcessingException e) {
+            throw CommonError.jsonOperationError("Common", map.toString(), e);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/deserialize/GoogleSheetsDeserializer.java]: ditto

File: seatunnel-connectors-v2/connector-google-sheets/src/test/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/deserialize/GoogleSheetsDeserializerTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                new JsonDeserializationSchema(false, false, schema);
+        final GoogleSheetsDeserializer googleSheetsDeser =
+                new GoogleSheetsDeserializer(schema.getFieldNames(), deser);
+        List<Object> row = new ArrayList<>();
+        Object mockObj = new Object();
+        row.add(mockObj);
+
+        String expectedPayload = String.format("{name=%s}", mockObj.toString());
+        SeaTunnelRuntimeException expected =
+                CommonError.jsonOperationError("Common", expectedPayload);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-google-sheets/src/test/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/deserialize/GoogleSheetsDeserializerTest.java]: ditto

**Review by Hisoka-X — COMMENTED** [review]
I think we should change json error msg from
```
JSON_OPERATION_FAILED("COMMON-02", "<format> JSON convert/parse '<payload>' operation failed.")
```
to
```
JSON_OPERATION_FAILED("COMMON-02", "<identifier> JSON convert/parse '<payload>' operation failed.")
```

**Comment by CheneyYin** [comment]
> I think we should change json error msg from
> ```
> JSON_OPERATION_FAILED("COMMON-02", "<format> JSON convert/parse '<payload>' operation failed.")
> ```
> to
> ```
> JSON_OPERATION_FAILED("COMMON-02", "<identifier> JSON convert/parse '<payload>' operation failed.")
> ```
> 

Agree

File: seatunnel-connectors-v2/connector-google-sheets/src/test/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/deserialize/GoogleSheetsDeserializerTest.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+        SeaTunnelRuntimeException actual =
+                assertThrows(
+                        SeaTunnelRuntimeException.class,
+                        () -> {
+                            googleSheetsDeser.deserializeRow(row);
+                        },
+                        "expecting exception message: " + expected.getMessage());
+
+        assertEquals(actual.getMessage(), expected.getMessage());
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-google-sheets/src/test/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/deserialize/GoogleSheetsDeserializerTest.java]:
```suggestion
        assertEquals(expected.getMessage(), actual.getMessage());
```

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 9]
```diff
@@ -82,65 +83,82 @@ public String serializeRow(SeaTunnelRow row) {
     private String serializeUpsert(SeaTunnelRow row) {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java:83]: Could you add a test case for this method? Thanks.

Comment (CheneyYin) [reply seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java:83]: ACK

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**